src/gluonts/model/deepar/_network.py

```diff
@@ -291,6 +295,26 @@ def unroll_encoder(
 
 
 class DeepARTrainingNetwork(DeepARNetwork):
+    @validated()
+    def __init__(self, alpha: float = 0, beta: float = 0, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        # regularization weights
+        self.alpha = alpha
+        self.beta = beta
+
+        if alpha:
+            self.ar_loss = ActivationRegularizationLoss(
+                alpha, time_axis=1, batch_axis=0
+            )
+        if beta:
+            self.tar_loss = TemporalActivationRegularizationLoss(
+                beta, time_axis=1, batch_axis=0
+            )
+
+        if alpha or beta:
+            self.rnn_outputs = None
```

Comment:
Don't we need to define `rnn_outputs` even if both `alpha` and `beta` are zero?
In any case `rnn_outputs` should not be here: this object defines the network in a static sense meaning what this network is composed of; `rnn_outputs` is a dynamic thing that depends on some input data.

Reply:
rnn_outputs needs to be cached only when regularizations are to be added. When both alpha and beta are zero, there will be no regularizations at all.

Reply:
If both `alpha` and `beta` are zero, then the constructor does not define the instance attribute `rnn_outputs`.
But you are creating the instance attribute `rnn_outputs` [here, outside of `__init__`](https://github.com/awslabs/gluon-ts/pull/955/files#diff-07d7aa1021a99ecb21d2c5024a21a4aaR364) irrespective of what `alpha` and `beta` are, right? This is not a good practice (although allowed in python).

In any case, as I mentioned in the comment, `rnn_outputs` should not belong here.

Reply:
I agree. I will add "return_rnn_outputs" with default value False to distribution, so that it won't affect its default behavior used in other places, and make it pass rnn_outputs when necessary.

Reply:
cool, thanks.